packages/hint-sri/src/hint.ts

```diff
@@ -296,17 +362,130 @@ Actual:   ${hashes.join(', ')}`;
+             * should report what we have in the cache after the
+             * first 'scan::end'.
+             */
+            if (this.scanEnded) {
+                const promises = this.getCache(evt).map((error) => {
+                    return this.context.report(error.resource, error.message, error.options);
+                });
+
+                await Promise.all(promises);
+            }
+
+            return;
+        }
+
+        const { resource, element } = evt;
+        let content: NetworkData;
+
+        /*
+         * If the resource is a local file, ignore the analysis.
+         * The sri usually is added on the building process before publish,
+         * so is going to be very common that the sri doesn't exists
+         * for local files.
+         */
+        if (resource.startsWith('file://')) {
+            debug(`Ignoring local resource: ${resource}`);
+
+            return;
+        }
+
+        /*
+         * `requestAsync` is not included in webpack bundle for `extension-browser`.
+         * This is ok because the browser will have already requested this via `fetch::end`
+         * events.
+         *
+         * Note: We are not using `Requester` becuase it depends on `iltorb` and it can
+         * cause problems with the vscode-extension because `iltorb` dependens on the
+         * node version for which it was compiled.
+         */
+        if (!requestAsync) {
+            return;
+        }
+
+        try {
+            content = {
+                request: {} as Request,
+                response: {
+                    body: {
+                        content: await requestAsync({
+                            method: 'GET',
+                            url: resource
+                        })
+                    }
+                } as Response
+            };
+        } catch (e) {
+            debug(`Error accessing ${resource}. ${JSON.stringify(e)}`);
+
+            await this.context.report(origin, `Can't get the resource ${resource}`, { element });
+
+            return;
+        }
+
+        await this.validateResource(Object.assign(evt, {
+            request: content.request,
+            response: content.response
+        }), origin);
+    }
+
+    /** Validation entry point for event element::script */
+    private async validateScriptElement(evt: ElementFound) {
+        const isScript = await this.isScriptOrLink(evt as FetchEnd);
+
+        if (!isScript) {
+            return;
+        }
+
+        const origin = evt.resource;
+
+        /*
+         * 'this.isScriptOrLink' has already checked
+         * that the src attribute exists, so it is safe to use !.
+         */
+        evt.resource = new URL(evt.element.getAttribute('src')!, evt.resource).href;
+        await this.validateElement(evt, origin);
+    }
+
+    /** Validation entry point for events element::link */
+    private async validateStyleElement(evt: ElementFound) {
```

Comment:
Can't this be combined with `validateScriptElement`?
Both seem to do the same thing and the only thing that changes is  the attribute to get (`src` or `href`).